### ISSUE TITLE
For removeLayerControls of layerControl

### ISSUE THREAD

**Issue by ValueCreation** [cmv/cmv-app#526]
```
// Add Layer
this.meshLayer = new FeatureLayer(arg.url, arg.layerOptions);
this.map.addLayers(featureLayers);

this.map.on("layer-add-result", function(args) {
    var layer = args.layer;
    var title = "test123;
    var layerControlInfo = {
              controlOptions:{ expanded:true, metadataUrl:false, swipe:false }, 
              layer: layer,
              title: title,
              type: 'feature'
   };
   topic.publish('layerControl/addLayerControls', [layerControlInfo]);
});

// removeLayer
this.map.removeLayer(this.meshLayer)
topic.publish('layerControl/removeLayerControls', [this.meshLayer]);
```
Additional layer is not a problem, but it is a error in the deletion of the layer.
It is an error in the map control.
Specifically, an error will occur in the zoom out and zoom in.
The accompanying error message in chrome.
[
![2016-03-31_1420](https://cloud.githubusercontent.com/assets/6479727/14166229/8813afac-f74c-11e5-8201-897cee9ca6d4.png)
](url)

Branch, uses the latest develop branch.
Could you tell me if such something workaround.
I feel you have as events of _Contorl.js of destroy () and map.removeLayer ().

**Comment by tmcgee** [comment]
@ValueCreation Can you provide a full example demonstrating the issue for me to review. Thanks in advance.

**Comment by ValueCreation** [comment]
@tmcgee 
The problem happens when you remove a layer.
You will not be able to scale the map.

Deleted layer will be the feature layer.

**Comment by tmcgee** [comment]
@ValueCreation I am hoping you will be able provide a simple example that demonstrates the issue for myself or others to test.

**Comment by ValueCreation** [comment]
The problem is....when I delete future layer, I will not be able to to use the function of map zoom-in/zoom-out.

**Comment by ValueCreation** [comment]
addLayer, removeLayer,  sample Widget. 
http://valuecreation.github.io/cmv-app/viewer/

**Comment by AlexAThomas** [comment]
@ValueCreation Is the problem resolved if you add a destroy method on the Feature.js widget?

/* you would need to remove the     'use strict';  line in this file. */

        destroy: function () {
            this.inherited(arguments);
        }

**Comment by tmcgee** [comment]
@AlexAThomas From a very quick test, adding that destroy method to Feature.js control does fix it.  Doe that mean we may have an issue with the destroy method you added to `_Control.js` which is mixed-in to Feature.js?

**Comment by ValueCreation** [comment]
![2016-04-06_0118](https://cloud.githubusercontent.com/assets/6479727/14289390/87392038-fb95-11e5-91b6-18996da287e5.png)

@AlexAThomas Thank you very much.
It could not be improved by adding the destroy method to Feature.js.
it's difficult.

**Comment by AlexAThomas** [comment]
@tmcgee i'm not sure yet... I will need to test my feature layer (after removing the destroy method) and see if the _Control.js destroy method is being called.

**Comment by AlexAThomas** [comment]
@ValueCreation remove the  "use strict" directive on line 16 and try the destroy

@tmcgee It looks like the call to the _Control.js destroy method working as intended (at least with a graphic layer) 

I added a graphic Layer as a feature layer (with no graphics though), then removed it.  The destroy() call was hit.

`var glayer = new esri.layers.GraphicsLayer({ id: "test" });
var gSymbol = new esri.symbol.PictureMarkerSymbol("images/rocket-logo.png", 434, 244);
var gRenderer = new esri.renderer.SimpleRenderer(gSymbol);
glayer.setRenderer(gRenderer);
this.app.map.addLayer(glayer);
var layerInfo = {layer : glayer, title: "My Sample Graphic", type : "feature"};
dojo.publish('layerControl/addLayerControls', [[layerInfo]])


dojo.publish("layerControl/removeLayerControls", [[glayer]])
`


thoughts?

**Comment by ValueCreation** [comment]
_Control.js 
this._scaleRangeHandler.remove();
this._scaleRangeHandler = null;
Add Once you have been resolved.
```
        destroy: function () {
            this._scaleRangeHandler.remove();
            this._scaleRangeHandler = null;

            this.inherited(arguments);

            this._handlers.forEach(function (h) {
                h.remove();
            });
        }
```

**Comment by AlexAThomas** [comment]
@ValueCreation good catch!  All the services I've been using have min/max scale set to 0.

**Comment by tmcgee** [comment]
@AlexAThomas @ValueCreation Should `this._scaleRangeHandler` be pushed into the `this._handlers` array when it is set so it will removed in the forEach loop in destroy method?

**Comment by AlexAThomas** [comment]
@tmcgee Created a [PR](https://github.com/cmv/cmv-app/pull/534) to add scaleRangeHandler to this._handlers.  This handles the following special case:

//console
var l = this.app.map.getLayer("DamageAssessment");
l.setMaxScale(130000);
l.setMaxScale(130001);
l.setMaxScale(0);

**Comment by tmcgee** [comment]
Fixed in #534
